src/node.ts

```diff
@@ -5,32 +5,180 @@ import { analyticsNode } from './plugins/analytics-node'
 import { Plugin } from './core/plugin'
 import { EventQueue } from './core/queue/event-queue'
 import { PriorityQueue } from './lib/priority-queue'
+import { Options } from './core/events/interfaces'
+import { Callback, DispatchedEvent } from './core/arguments-resolver'
+import { Emitter } from './core/emitter'
+import { UserOptions } from './core/user'
 
-export class AnalyticsNode {
-  static async load(settings: {
-    writeKey: string
-  }): Promise<[Analytics, Context]> {
-    const cookieOptions = {
-      persist: false,
-    }
-
-    const queue = new EventQueue(new PriorityQueue(3, []))
-    const options = { user: cookieOptions, group: cookieOptions }
-    const analytics = new Analytics(settings, options, queue)
-
-    const nodeSettings = {
-      writeKey: settings.writeKey,
-      name: 'analytics-node-next',
-      type: 'after' as Plugin['type'],
-      version: 'latest',
-    }
-
-    const ctx = await analytics.register(
-      validation,
-      analyticsNode(nodeSettings)
-    )
-    analytics.emit('initialize', settings, cookieOptions ?? {})
-
-    return [analytics, ctx]
+export type Identity =
+  | { userId: string; anonymousId?: string }
+  | { userId?: string; anonymousId: string }
+
+export type NodeOptions = Options & Identity
+
+export interface AnalyticsNodeJs extends Emitter {
+  /**
+   * Combines two unassociated user identities.
+   * @param userId - The new user id you want to associate with the user.
+   * @param previousId - The previous id that the user was recognized by.
+   * @param options
+   * @param callback
+   */
+  alias(
+    userId: string,
+    previousId: string,
+    options?: Options,
+    callback?: Callback
+  ): Promise<DispatchedEvent>
+
+  /**
+   * Associates an identified user with a collective.
+   * @param groupId - The group id to associate with the provided user.
+   * @param traits - A dictionary of traits for the group.
+   * @param options - A dictionary of options including the user id.
+   * @param callback
+   */
+  group(
+    groupId: string,
+    traits: object,
+    options: NodeOptions,
+    callback?: Callback
+  ): Promise<DispatchedEvent>
+
+  /**
+   * Includes a unique userId and/or anonymousId and any optional traits you know about them.
+   * @param userId
+   * @param traits
+   * @param options
+   * @param callback
+   */
+  identify(
+    userId: string,
+    traits?: object,
+    options?: Options,
+    callback?: Callback
+  ): Promise<DispatchedEvent>
+
+  /**
+   * Records page views on your website, along with optional extra information
+   * about the page viewed by the user.
+   * @param properties
+   * @param options
+   * @param callback
+   */
+  page(
+    properties: object,
```

Comment:
The docstring says this is a dictionary. If so, wouldn’t  the correct type be Record<string | number, unknown> or something like that? object includes Array, null, etc IIRC.

Reply:
That sounds reasonable! I copied the type from what `Analytics` is using but might as well tighten the types up here!

Reply:
ah K, figured. We might want to take this opportunity to create an official "Properties" or "PropertiesDictionary" type, since we use it all over the app